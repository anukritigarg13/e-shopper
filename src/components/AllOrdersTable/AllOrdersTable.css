.orders-table{
    margin: 5rem 5rem;
}

.order-desc-table thead{
    background-color: rgb(27, 140, 160);
}
.all-orders-items-table{
    overflow-x: scroll;
}
