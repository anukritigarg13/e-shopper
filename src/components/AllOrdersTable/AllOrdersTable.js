import React from 'react';
import PropTypes from 'prop-types';
import OrderDescTable from '../OrderDescriptionTable/OrderDescriptionTable';
import './AllOrdersTable.css';

const AllOrdersTable = ({
  id, order, date, totalCost, totalItems,
}) => {
  const formattedDate = new Date(date).toDateString();
  console.log(formattedDate);
  return (
    <div className="orders-table">
      <table className="all-orders-table">
        <thead>
          <tr>
            <th>ORDER</th>
            <th>ITEMS</th>
            <th>DATE</th>
            <th>AMOUNT</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{id}</td>
            <td>{totalItems}</td>
            <td>
              {formattedDate}
            </td>
            <td>{totalCost}</td>
          </tr>
        </tbody>
      </table>
      <div className="order-desc-table">
        <OrderDescTable products={order} />
      </div>

    </div>
  );
};
const productsShape = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
});
AllOrdersTable.propTypes = {
  order: PropTypes.objectOf(PropTypes.arrayOf(productsShape)).isRequired,
  date: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
  totalItems: PropTypes.number.isRequired,
  totalCost: PropTypes.number.isRequired,
};
export default AllOrdersTable;
