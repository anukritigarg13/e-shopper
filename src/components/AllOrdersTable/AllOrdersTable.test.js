/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import AllOrdersTable from './AllOrdersTable';

describe('AllOrdersTable component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  let mockProps;
  beforeAll(() => {
    mockProps = {
      order: {
        Fruits: [{
          id: 1, name: 'MOCK', price: 40, count: 5, category: 'Fruits',
        }],
      },
      date: 1615447258012,
      id: 1,
      totalItems: 1,
      totalCost: 340,
    };
  });
  it('should render one OrderDescriptionTable component', () => {
    render(<AllOrdersTable {...mockProps} />);
    const orderTableElements = screen.getAllByTestId('order-desc-table');
    expect(orderTableElements.length).toBe(1);
  });
  it('should match snapshot', () => {
    const { container } = render(<AllOrdersTable {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
