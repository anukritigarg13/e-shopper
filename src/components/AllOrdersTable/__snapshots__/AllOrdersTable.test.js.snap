// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrdersTable component should match snapshot 1`] = `
<div>
  <div
    class="orders-table"
    data-testid="orders-table-test"
  >
    <table
      class="all-orders-table"
    >
      <thead>
        <tr>
          <th>
            ORDER
          </th>
          <th>
            ITEMS
          </th>
          <th>
            DATE
          </th>
          <th>
            AMOUNT
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            1
          </td>
          <td>
            Thu Mar 11 2021
          </td>
          <td>
            340
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="order-desc-table"
    >
      <table
        data-testid="order-desc-table"
      >
        <thead>
          <tr>
            <th>
              ITEM DESCRIPTION
            </th>
            <th>
              UNIT PRICE
            </th>
            <th>
              QUANTITY
            </th>
            <th>
              SUBTOTAL
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="category-row"
              colspan="4"
            >
              Fruits
            </td>
          </tr>
          <tr>
            <td>
              MOCK
            </td>
            <td>
              40
            </td>
            <td>
              5
            </td>
            <td>
              200
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
